### PR TITLE
linters: Bump golangci-lint version to v1.52.2 and fix issues

### PR DESCRIPTION
**Pull request by samuael**
This pull request contains several fixes to the gocryptotrader codebase, including a fix for a linter issue, suppressing some linter issues by adding a linter directive, removing unnecessary linter directives, and changing some variable names to constants.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as the item is complete.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [X] go test ./... -race
- [X] golangci-lint run
- [ ] Test X

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/bitfinex/bitfinex_test.go [hunk 1]
```diff
@@ -1085,7 +1085,7 @@ func TestWsAuth(t *testing.T) {
 	runAuth(t)
 }
 
-func runAuth(t *testing.T) {
+func runAuth(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline exchanges/bitfinex/bitfinex_test.go:1088]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: currency/forexprovider/fixer.io/fixer_test.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ var f Fixer
 
 var isSetup bool
 
-func setup(t *testing.T) {
+func setup(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline currency/forexprovider/fixer.io/fixer_test.go:16]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: database/repository/audit/audit_test.go [hunk 3]
```diff
@@ -99,7 +99,7 @@ func TestAudit(t *testing.T) {
 	}
 }
 
-func writeAudit(t *testing.T) {
+func writeAudit(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline database/repository/audit/audit_test.go:102]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: database/repository/audit/audit_test.go [hunk 4]
```diff
@@ -116,7 +116,7 @@ func writeAudit(t *testing.T) {
 	wg.Wait()
 }
 
-func readHelper(t *testing.T) {
+func readHelper(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline database/repository/audit/audit_test.go:119]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: database/repository/withdraw/withdraw_test.go [hunk 5]
```diff
@@ -153,7 +153,7 @@ func seedWithdrawData() {
 		Event(resp)
 	}
 }
-func withdrawHelper(t *testing.T) {
+func withdrawHelper(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline database/repository/withdraw/withdraw_test.go:156]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: engine/database_connection_test.go [hunk 6]
```diff
@@ -10,7 +10,7 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/database/drivers"
 )
 
-func CreateDatabase(t *testing.T) {
+func CreateDatabase(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline engine/database_connection_test.go:13]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/bittrex/bittrex_test.go [hunk 7]
```diff
@@ -604,7 +604,7 @@ func TestModifyOrder(t *testing.T) {
 	}
 }
 
-func WithdrawCryptocurrencyFunds(t *testing.T) {
+func WithdrawCryptocurrencyFunds(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline exchanges/bittrex/bittrex_test.go:607]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/hitbtc/hitbtc_test.go [hunk 8]
```diff
@@ -467,7 +467,7 @@ func TestGetDepositAddress(t *testing.T) {
 		}
 	}
 }
-func setupWsAuth(t *testing.T) {
+func setupWsAuth(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline exchanges/hitbtc/hitbtc_test.go:470]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/kline/kline_test.go [hunk 9]
```diff
@@ -877,8 +877,7 @@ func BenchmarkJustifyIntervalTimeStoringUnixValues1(b *testing.B) {
 	tt2 := time.Now().Add(-time.Hour)
 	tt3 := time.Now().Add(time.Hour)
 	for i := 0; i < b.N; i++ {
-		if tt1.Unix() == tt2.Unix() || //nolint:staticcheck // it is a benchmark to demonstrate inefficiency in calling
-			(tt1.Unix() > tt2.Unix() && tt1.Unix() < tt3.Unix()) {
+		if tt1.Unix() == tt2.Unix() || (tt1.Unix() > tt2.Unix() && tt1.Unix() < tt3.Unix()) { //nolint
```
Comment (gloriousCode) [inline exchanges/kline/kline_test.go:880]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/kline/kline_test.go [hunk 10]
```diff
@@ -893,7 +892,7 @@ func BenchmarkJustifyIntervalTimeStoringUnixValues2(b *testing.B) {
 	tt2 := time.Now().Add(-time.Hour).Unix()
 	tt3 := time.Now().Add(time.Hour).Unix()
 	for i := 0; i < b.N; i++ {
-		if tt1 >= tt2 && tt1 <= tt3 { //nolint:staticcheck // it is a benchmark to demonstrate inefficiency in calling
+		if tt1 >= tt2 && tt1 <= tt3 { //nolint
```
Comment (gloriousCode) [inline exchanges/kline/kline_test.go:895]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/stream/buffer/buffer_test.go [hunk 11]
```diff
@@ -47,7 +47,7 @@ func createSnapshot() (holder *Orderbook, asks, bids orderbook.Items, err error)
 
 	ch := make(chan interface{})
 	go func(<-chan interface{}) { // reader
-		for range ch {
+		for range ch { //nolint
```
Comment (gloriousCode) [inline exchanges/stream/buffer/buffer_test.go:50]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

File: exchanges/zb/zb_test.go [hunk 12]
```diff
@@ -37,7 +37,7 @@ func areTestAPIKeysSet() bool {
 	return z.ValidateAPICredentials(z.GetDefaultCredentials()) == nil
 }
 
-func setupWsAuth(t *testing.T) {
+func setupWsAuth(t *testing.T) { //nolint
```
Comment (gloriousCode) [inline exchanges/zb/zb_test.go:40]: Please be more specific with your nolint usage. Here is an example from our codebase:
`//nolint:forcetypeassert // Not necessary from a pool`
we specify what we are nolinting and we provide a comment on why we are nolinting

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Always nice to see you get more familiar with our codebase 🎉 
You'll just need to be more specific with your nolints

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks for the changes! Are you getting these linter issues from the most recent version? If so, can you please bump all versions for GitHub actions like `linter.yml` and else ware?

**Comment by samuael** [comment]
Yes, I found these issues after upgrading my linter. golangci-lint v1.52.2 with go1.20.2.

Okay sir I will update the .github/workflows/linter.yml file with my next commit. After few minutes

**Review by shazbert — CHANGES_REQUESTED** [review]
Appveyor is complaining :

image


Could you please bump the versions for golanci-lint across the code base.

**Review by shazbert — APPROVED** [review]
This is lovely. Thanks! 👍

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes

**Review by thrasher- — APPROVED** [review]
Nice work @samuael !

**Comment by samuael** [comment]
Thank you, sir.
and I want to give a big Thanks to @shazbert  and @gloriousCode for your help on this PR.

**Pull request merged**